heka/plugins/s3splitfile/s3splitfile_output.go

```diff
@@ -114,19 +116,21 @@ const (
 
 func (o *S3SplitFileOutput) ConfigStruct() interface{} {
 	return &S3SplitFileOutputConfig{
-		Perm:           "644",
-		FlushInterval:  1000,
-		FolderPerm:     "700",
-		MaxFileSize:    524288000,
-		MaxFileAge:     3600000,
-		MaxOpenFiles:   1000,
-		AWSKey:         "",
-		AWSSecretKey:   "",
-		AWSRegion:      "us-west-2",
-		S3Bucket:       "",
-		S3BucketPrefix: "",
-		S3Retries:      5,
-		S3WorkerCount:  10,
+		Perm:             "644",
+		FlushInterval:    1000,
+		FolderPerm:       "700",
+		MaxFileSize:      524288000,
+		MaxFileAge:       3600000,
+		MaxOpenFiles:     1000,
+		AWSKey:           "",
+		AWSSecretKey:     "",
+		AWSRegion:        "us-west-2",
+		S3Bucket:         "",
+		S3BucketPrefix:   "",
+		S3Retries:        5,
+		S3ConnectTimeout: 300,
```

Comment:
Failures in S3Output are far more important than S3Input. We want to wait as long as possible before declaring failure. There is actual retry logic in the output (where that is not implemented in the input yet), so this should mostly be mitigated by retries, but tl;dr we want to try harder on Output.